cookiecutter/prompt.py

```diff
@@ -17,21 +17,88 @@
 from jinja2.environment import Environment
 
 
-def read_response(prompt=''):
-    """Prompt the user and return the entered value or an empty string.
+def read_user_variable(var_name, default_value):
+    """Prompt the user for the given variable and return the entered value
+    or the given default.
 
-    :param str prompt: Text to display to the user
+    :param str var_name: Variable of the context to query the user
+    :param default_value: Value that will be returned if no input happens
+    """
+    # Please see http://click.pocoo.org/4/api/#click.prompt
+    return click.prompt(var_name, default=default_value)
+
+
+def read_user_yes_no(question, default_value):
+    """Prompt the user to reply with 'yes' or 'no' (or equivalent values).
+
+    Note:
+      Possible choices are 'true', '1', 'yes', 'y' or 'false', '0', 'no', 'n'
+
+    :param str question: Question to the user
+    :param default_value: Value that will be returned if no input happens
     """
     # Please see http://click.pocoo.org/4/api/#click.prompt
-    # default: use an empty string if no input happens
-    # prompt_suffix: do not add a suffix to the prompt
-    # show_default: the prompt is expected to mention the default itself
     return click.prompt(
-        prompt,
-        default='',
-        prompt_suffix='',
-        show_default=False,
+        question,
+        default=default_value,
+        type=click.BOOL
+    )
+
+
+def read_user_choice(var_name, options):
+    """Prompt the user to choose from several options for the given variable.
+
+    The first item will be returned if no input happens.
+
+    :param str var_name: Variable as specified in the context
+    :param list options: Sequence of options that are available to select from
+    :return: Exactly one item of ``options`` that has been chosen by the user
+    """
+    # Please see http://click.pocoo.org/4/api/#click.prompt
+    if not isinstance(options, list):
+        raise TypeError
+
+    if not options:
+        raise ValueError
+
+    if not all(is_string(opt) for opt in options):
+        raise TypeError
+
+    choice_map = OrderedDict(
+        ('{}'.format(i), value) for i, value in enumerate(options, 1)
     )
+    choices = choice_map.keys()
+    default = '1'
+
+    choice_lines = ['{} - {}'.format(*c) for c in choice_map.items()]
+    prompt = '\n'.join((
+        'Select {}:'.format(var_name),
+        '\n'.join(choice_lines),
+        'Choose from {}'.format(', '.join(choices))
+    ))
+
+    user_choice = click.prompt(
+        prompt, type=click.Choice(choices), default=default
+    )
+    return choice_map[user_choice]
+
+
+def _render_variable(env, raw, cookiecutter_dict):
+    raw = raw if is_string(raw) else str(raw)
```

Comment:
I'm not a fan of ternary operators as they are not as clear as breaking it up into multiple lines of code.